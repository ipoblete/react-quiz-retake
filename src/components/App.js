import React, { Component } from 'react';
import Controls from './Controls';
import Display from './Display';
import 'normalize.css';
import '../css/App.css';

export default class App extends Component {
  state = {
    title: 'React',
    subtitle: 'Make-up Quiz'
  };

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  };

  render() {
    const { title, subtitle } = this.state;
    return (
      <>
        <h1>REACT MAKE-UP!</h1>
        <Controls 
          title={title}
          subtitle={subtitle}
          onChange={this.handleChange}
        />

        <Display 
          title={title}
          subtitle={subtitle}
        />
      </>
    );
  }
}
