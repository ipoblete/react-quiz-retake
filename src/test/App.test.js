import React from 'react';
import renderer from 'react-test-renderer';
import App from '../components/App';

describe('App', () => {
  it('matches a snapshot', () => {
    const app = renderer.create(
      <App />
    ).toJSON();

    expect(app).toMatchSnapshot();
  });
});
