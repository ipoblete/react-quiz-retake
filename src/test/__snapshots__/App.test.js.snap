// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches a snapshot 1`] = `
Array [
  <h1>
    REACT MAKE-UP!
  </h1>,
  <form>
    <label>
      Title:
    </label>
    <input
      name="title"
      onChange={[Function]}
      type="text"
      value="React"
    />
    <label>
      Subtitle:
    </label>
    <input
      name="subtitle"
      onChange={[Function]}
      type="text"
      value="Make-up Quiz"
    />
  </form>,
  "React",
  "Make-up Quiz",
]
`;
